src/core/main_task.c

```diff
@@ -372,8 +407,19 @@ void MainTask(void *pvParameters)
         /*
          * 13. EEPROM total operating writing
          */
-
-        // TODO
+        if (xTaskGetTickCount() - write_time > pdMS_TO_TICKS(WRITE_EEPROM_PERIOD_MS)) {
+            write_time = xTaskGetTickCount();
+            total_operating_time += WRITE_EEPROM_PERIOD_MS/1000L;
+            incrementCyclesNumber(); // FIXME Remove this (should be called by another task)
+#if DEBUG_MAIN
+            debug_print("[MAIN] EEPROM, tot: %u\r\n", total_operating_time);
+            debug_print("[MAIN] EEPROM, cn: %u\r\n", cycles_number);
+#endif
+            taskENTER_CRITICAL();
+            eeprom_write_dword (&operating_time_addr, total_operating_time);
+            eeprom_write_dword (&cycles_number_addr, cycles_number);
+            taskEXIT_CRITICAL();
```

Comment:
Could this be split into two distinct critical sections ?
```suggestion
            taskENTER_CRITICAL();
            eeprom_write_dword (&operating_time_addr, total_operating_time);
            taskEXIT_CRITICAL();
            taskENTER_CRITICAL();
            eeprom_write_dword (&cycles_number_addr, cycles_number);
            taskEXIT_CRITICAL();
```